ntplib.py

```diff
@@ -280,6 +281,8 @@ def dest_time(self):
 class NTPClient(object):
     """NTP client session."""
 
+    _MAX_HOSTS_TO_TRY = 8 # Normally pool responses are 4 hosts and hence a max of 8 v4+v6 addresses; but other hosts could be higher
+
```

Comment:
I'm not a fan of this: I don't believe a DNS response could have enough entries for it to be a problem in practice.

Reply:
This was defined out of an abundance of caution such that should a DNS response be massive, the `random.sample()` call doesn't work on and return a massive dataset. To be honest, if `_MAX_HOSTS_TO_TRY` was 1000, the code would still be near-instant. I'm happy to up that value above `8`; however, keep in mind that the current code operates with `[0]` and hence even a max value of 1 provides a similar logic and results. Going back and looking at the NTP pool responses, I only see eight address values being returned.